### PR TITLE
Added classes with HW34

### PR DESCRIPTION
**Pull request by Dimidow**

### CODE REVIEW

File: HW34/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>6.0.0</version>
+        </dependency>
+
+        <dependency>
```
Comment (serhii-tokranov) [inline HW34/pom.xml]: spring-boot dependency used to early :)

File: HW34/src/main/java/org/my/homework/ProductRepository.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.List;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductRepository {
+
+    @Bean
+    public List<Product> getAllProductsList() {
```
Comment (serhii-tokranov) [inline HW34/src/main/java/org/my/homework/ProductRepository.java]: Oh no, this is not a bean... 
The ProductRepository is a spring component (in other words, a bean). This component manages products (something like Facad, which hides the persistence implementation). So, there is no sense making List of Products a bean - ProductRepository is responsible for it.

File: HW34/src/main/java/org/my/homework/Cart.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+@Scope("prototype")
+public class Cart {
+
+    @Autowired
+    private List<Product> products;
```
Comment (serhii-tokranov) [inline HW34/src/main/java/org/my/homework/Cart.java]: This is not a bean - no autowired and in general the logic that stands behind this List must be in ProductRepository

File: HW34/src/main/java/org/my/homework/ConsoleCartApplication.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class ConsoleCartApplication {
+
+    @Autowired
+    private ConsoleCartService consoleCartService;
+
+    public static void main(String[] args) {
+        SpringApplication.run(ConsoleCartApplication.class, args);
```
Comment (serhii-tokranov) [inline HW34/src/main/java/org/my/homework/ConsoleCartApplication.java]: For this task should not be used Spring boot.

File: HW34/src/main/java/org/my/homework/ProductRepository.java (hunk 2 above)
Comment (Dimidow) [reply HW34/src/main/java/org/my/homework/ProductRepository.java]: Not sure that a new realisation is much better but here was fixed

File: HW34/src/main/java/org/my/homework/ConsoleCartApplication.java (hunk 4 above)
Comment (Dimidow) [reply HW34/src/main/java/org/my/homework/ConsoleCartApplication.java]: Fixed. The starter was implemented by AnnotationConfigApplicationContext

File: HW34/src/main/java/org/my/homework/configuration/MyConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public Cart cart() {
+        return new Cart(productRepository());
+    }
+
+    @Bean
+    public ConsoleCartService consoleCartService(){
+        return new ConsoleCartService(cart());
+    }
+
+    private List<Product> products(){
```
Comment (serhii-tokranov) [inline HW34/src/main/java/org/my/homework/configuration/MyConfiguration.java:34]: Nice

File: HW34/src/main/java/org/my/homework/configuration/MyConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+@Configuration
+public class MyConfiguration {
+
+    @Bean
+    public ProductRepository productRepository() {
+        return new ProductRepository(products());
+    }
+
+    @Bean
+    public Cart cart() {
```
Comment (serhii-tokranov) [inline HW34/src/main/java/org/my/homework/configuration/MyConfiguration.java:25]: You have 2 different variants of creating the Cart bean. Here programmatically and via annotations (@Component). It is not correct. You have to use only one way to create a Cart. Plus, the programmatically variant has a singleton scope. Meanwhile, the annotation variant has a prototype scope.

Comment (Dimidow) [reply HW34/src/main/java/org/my/homework/configuration/MyConfiguration.java:25]: Fixed

**Pull request merged**